src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CollectionCodecs.java

```diff
@@ -103,14 +132,38 @@ static Set<Object> toCQLSet(
         continue;
       }
       if (elementCodec == null || !elementCodec.handlesJavaValue(element)) {
-        elementCodec = findElementCodec(valueCodecs, elementType, element);
+        elementCodec = findCollectionElementCodec(valueCodecs, elementType, element);
       }
       result.add(elementCodec.toCQL(element));
     }
     return result;
   }
 
-  private static JSONCodec<Object, Object> findElementCodec(
+  static Map<String, Object> toCQLMap(
+      List<JSONCodec<?, ?>> valueCodecs, DataType elementType, Map<?, ?> rawMapValue)
+      throws ToCQLCodecException {
+    Map<String, JsonLiteral<?>> mapValue = (Map<String, JsonLiteral<?>>) rawMapValue;
+    Map<String, Object> result = new LinkedHashMap<>(mapValue.size());
+    JSONCodec<Object, Object> elementCodec = null;
+    for (Map.Entry<String, JsonLiteral<?>> entry : mapValue.entrySet()) {
+      String key = entry.getKey();
+      Object element = entry.getValue().value();
+      if (element == null) {
+        result.put(key, null);
+        continue;
+      }
+      // In most cases same codec can be used for all elements, but we still need to check
+      // since same CQL value type can have multiple codecs based on JSON value type
+      // (like multiple Number representations; or simple Text vs EJSON-wrapper)
+      if (elementCodec == null || !elementCodec.handlesJavaValue(element)) {
```

Comment:
Is there any value in caching previously seen elementCodec ? Or do we think this is rare enough (needing a diff codec) it's not worth it, or the cost is low  ?

Reply:
We probably should look into bit better caching once codecs are all there; for now keeping track of last `elementCodec` here should work (we will have more than one codecs for Number types, f.ex, since we have 3 from-JSON input number types to convert; similarly for Vectors we have 2 input types).